.github/workflows/compute-incremental-coverage.py

```diff
@@ -81,30 +116,38 @@ def __init__(self, file_data):
       #       "column": 0
       #     },
       #     "end": {
-      #       "line": 7,
+      #       "line": 8,
       #       "column": 29
-      #     },
-      #   },
-      #   "1": {
-      #     "start": {
-      #       "line": 9,
-      #       "column": 0
-      #     },
-      #     "end": {
-      #       "line": 10,
-      #       "column": 22
-      #     },
+      #     }
       #   },
       #   ...
       # }
-      # We first convert this into a set of lines with instrumentation.
       for key, value in path_data["statementMap"].items():
-        statement_to_lines[key] = []
-
-        start_line = value["start"]["line"]
-        end_line = value["end"]["line"]
-        for line in range(start_line, end_line + 1):
-          statement_to_lines[key].append(line)
+        # All the lines of the statement, which may include other functions or
+        # statements.
+        lines = CoverageLines(value)
+
+        # Subtract from that the lines of any function that is a subset of
+        # these lines.  By excluding entire methods before adding back their
+        # child statements, we exclude empty lines in class methods.
+        for function_lines in function_locations:
+          if function_lines < lines:  # strict subset
+            lines -= function_lines  # set subtraction
+
+        # If this statement is inside the range of another statement, remove
+        # this inner range from that outer one.  This is important because loops
+        # and conditional statements contain their inner branches.
+        for older_key, older_lines in statement_to_lines.items():
```

Comment:
Is it guaranteed that the inner statement always comes after the outer statement in the output?

Reply:
Yes, as far as I can tell.  Statements are in order by starting position.